src/main/java/Joey.java

```diff
@@ -0,0 +1,218 @@
+import java.util.Scanner;
+
+class Task {
+    private String description;
+    private boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public boolean isDone() {
+        return isDone;
+    }
+
+    public void markDone() {
+        isDone = true;
+    }
+
+    public void markNotDone() {
+        isDone = false;
+    }
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]");
+    }
+
+    public String getType() {
+        return "";
+    }
+
+    public String toString() {
+        return getType() + " " + getStatusIcon() + " " + description;
+    }
+}
+
+//class Todo extends Task {
+//    public Todo(String description) {
+//        super(description);
+//    }
+//
+//    public String getType() {
+//        return "[T]";
+//    }
+//}
+
+class Todo extends Task {
+    public Todo(String description) {
+        super(description);
+    }
+
+    public String getType() {
+        return "[T]";
+    }
+}
+
+
+//class Deadline extends Task {
+//    private String by;
+//
+//    public Deadline(String description, String by) {
+//        super(description);
+//        this.by = by;
+//    }
+//
+//    public String getType() {
+//        return "[D]";
+//    }
+//
+//    public String toString() {
+//        return super.toString() + " (by: " + by + ")";
+//    }
+//}
+
+class Deadline extends Task {
+    private String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public String getType() {
+        return "[D]";
+    }
+}
+
+
+//class Event extends Task {
+//    private String from;
+//    private String to;
+//
+//    public Event(String description, String from, String to) {
+//        super(description);
+//        this.from = from;
+//        this.to = to;
+//    }
+//
+//    public String getType() {
+//        return "[E]";
+//    }
+//
+//    public String toString() {
+//        return getType() + getStatusIcon() + " " + super.toString() + " (from: " + from + " to: " + to + ")";
+//    }
+//}
+
+class Event extends Task {
+    private String from;
+    private String to;
+
+    public Event(String description, String from, String to) {
+        super(description);
+        this.from = from;
+        this.to = to;
+    }
+
+    public String getType() {
+        return "[E]";
+    }
+
+    public String toString() {
+        return super.toString() + " (from: " + from + " to: " + to + ")";
+    }
+}
+
+
+public class Joey {
+    public static void main(String[] args) {
```

Comment:
The main method could be much more readable if it were broken down into smaller sub-functions. One such function could be called within each if-condition, given that they share much of the same logic.